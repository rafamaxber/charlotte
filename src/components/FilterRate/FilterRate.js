import React, { Component } from 'react';
import styled from 'styled-components';
import Rating from 'react-rating';
import starFullImage from '../../assets/star-filled.svg';
import starEmptyImage from '../../assets/star-outline.svg';

const WrapperFilter = styled.div`
  max-width: 300px;
  border-top: 1px solid #d9d9d9;
  margin-top: 27px;
  padding-top: 15px;
`;

const StarFull = styled.img.attrs({
  src: starFullImage,
})`
  width: 26px;
  margin-right: 4px;
`;

const StarEmpty = StarFull.extend.attrs({
  src: starEmptyImage,
})``;

const TextSmall = styled.div`
  font-family: 'Heebo', sans-serif;
  font-style: normal;
  font-weight: normal;
  line-height: normal;
  font-size: 12.8px;
  color: #B5B5B5;
  margin-bottom: 15px;
`;

const FilterRate = styled.div``;

export default class WrapperFilterRate extends Component {
  constructor(props) {
    super(props);
    this.state = this.getInitialState();
  }

  getInitialState() {
    return {
      min: 0,
      max: 5,
      default: 3,
      selected: 4,
    }
  }

  updateRating(rate) {
    console.log(rate)
  }

  render() {
    return(
      <WrapperFilter>
        <FilterRate>
          <TextSmall>
            Stars
          </TextSmall>
          <Rating
            stop={this.state.max}
            emptySymbol={<StarEmpty />}
            fullSymbol={<StarFull />}
            onChange={this.updateRating.bind(this)}
          />
        </FilterRate>
      </WrapperFilter>
    );
  }
}
