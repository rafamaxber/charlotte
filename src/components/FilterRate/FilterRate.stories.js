import React from 'react';
import { storiesOf } from '@storybook/react';
import FilterRate from './FilterRate';

storiesOf('FilterRate', module)
  .add('FilterRate', () => (
    <FilterRate>
    </FilterRate>
  ));
