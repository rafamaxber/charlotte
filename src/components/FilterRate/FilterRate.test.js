import React from 'react';
import renderer from 'react-test-renderer';
import FilterRate from './FilterRate';

describe('FilterRate', () => {
  it('Should match with snapshot', () => {
    const component = renderer.create(
      <FilterRate></FilterRate>
    );
    expect(component).toMatchSnapshot();
  });
});
