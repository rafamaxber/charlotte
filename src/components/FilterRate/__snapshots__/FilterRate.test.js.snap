// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FilterRate Should match with snapshot 1`] = `
<div
  className="sc-bdVaJa iDsbrN"
>
  <div
    className="sc-ifAKCX cFlEyZ"
  >
    <div
      className="sc-bxivhb iDpwYV"
    >
      Stars
    </div>
    <span
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
      style={
        Object {
          "direction": "ltr",
          "display": "inline-block",
        }
      }
    >
      <span
        onClick={[Function]}
        onMouseMove={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        style={
          Object {
            "cursor": "pointer",
            "display": "inline-block",
            "position": "relative",
          }
        }
      >
        <img
          className="sc-htpNat eUBtfO"
          src="star-outline.svg"
        />
        <span
          style={
            Object {
              "display": "inline-block",
              "left": 0,
              "overflow": "hidden",
              "position": "absolute",
              "top": 0,
              "width": "0%",
            }
          }
        >
          <span
            style={
              Object {
                "backgroundColor": "red",
                "border": "5px double white",
                "borderRadius": "50%",
                "display": "inline-block",
                "height": 30,
                "width": 30,
              }
            }
          />
        </span>
      </span>
      <span
        onClick={[Function]}
        onMouseMove={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        style={
          Object {
            "cursor": "pointer",
            "display": "inline-block",
            "position": "relative",
          }
        }
      >
        <img
          className="sc-htpNat eUBtfO"
          src="star-outline.svg"
        />
        <span
          style={
            Object {
              "display": "inline-block",
              "left": 0,
              "overflow": "hidden",
              "position": "absolute",
              "top": 0,
              "width": "0%",
            }
          }
        >
          <span
            style={
              Object {
                "backgroundColor": "red",
                "border": "5px double white",
                "borderRadius": "50%",
                "display": "inline-block",
                "height": 30,
                "width": 30,
              }
            }
          />
        </span>
      </span>
      <span
        onClick={[Function]}
        onMouseMove={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        style={
          Object {
            "cursor": "pointer",
            "display": "inline-block",
            "position": "relative",
          }
        }
      >
        <img
          className="sc-htpNat eUBtfO"
          src="star-outline.svg"
        />
        <span
          style={
            Object {
              "display": "inline-block",
              "left": 0,
              "overflow": "hidden",
              "position": "absolute",
              "top": 0,
              "width": "0%",
            }
          }
        >
          <span
            style={
              Object {
                "backgroundColor": "red",
                "border": "5px double white",
                "borderRadius": "50%",
                "display": "inline-block",
                "height": 30,
                "width": 30,
              }
            }
          />
        </span>
      </span>
      <span
        onClick={[Function]}
        onMouseMove={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        style={
          Object {
            "cursor": "pointer",
            "display": "inline-block",
            "position": "relative",
          }
        }
      >
        <img
          className="sc-htpNat eUBtfO"
          src="star-outline.svg"
        />
        <span
          style={
            Object {
              "display": "inline-block",
              "left": 0,
              "overflow": "hidden",
              "position": "absolute",
              "top": 0,
              "width": "0%",
            }
          }
        >
          <span
            style={
              Object {
                "backgroundColor": "red",
                "border": "5px double white",
                "borderRadius": "50%",
                "display": "inline-block",
                "height": 30,
                "width": 30,
              }
            }
          />
        </span>
      </span>
      <span
        onClick={[Function]}
        onMouseMove={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        style={
          Object {
            "cursor": "pointer",
            "display": "inline-block",
            "position": "relative",
          }
        }
      >
        <img
          className="sc-htpNat eUBtfO"
          src="star-outline.svg"
        />
        <span
          style={
            Object {
              "display": "inline-block",
              "left": 0,
              "overflow": "hidden",
              "position": "absolute",
              "top": 0,
              "width": "0%",
            }
          }
        >
          <span
            style={
              Object {
                "backgroundColor": "red",
                "border": "5px double white",
                "borderRadius": "50%",
                "display": "inline-block",
                "height": 30,
                "width": 30,
              }
            }
          />
        </span>
      </span>
    </span>
  </div>
</div>
`;
